orttraining/orttraining/python/training/utils/hooks/_subscriber_manager.py

```diff
@@ -0,0 +1,369 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+
+from collections import abc
+from typing import Callable, List, Union
+
+import warnings
+import torch
+
+from onnxruntime.training.ortmodule import ORTModule
+
+
+class ModuleHookSubscriberBase:
```

Comment:
Do you think we can move this class into a file of its own? The remained of this file is implementation of subscriber.
Also, do you think we can give it a simpler name?